packages/engine/lib/error/src/report.rs

```diff
@@ -0,0 +1,310 @@
+pub(super) use alloc::boxed::Box;
+use core::{fmt, fmt::Formatter, marker::PhantomData, panic::Location};
+#[cfg(feature = "backtrace")]
+use std::backtrace::{Backtrace, BacktraceStatus};
+#[cfg(feature = "std")]
+use std::error::Error as StdError;
+
+use provider::{
+    tags::{Ref, Value},
+    Provider, TypeTag,
+};
+#[cfg(feature = "spantrace")]
+use tracing_error::{SpanTrace, SpanTraceStatus};
+
+use super::Frame;
+use crate::{tags, Frames, Report, Requests};
+
+pub(super) struct ReportImpl {
+    pub(super) frame: Frame,
+    #[cfg(feature = "backtrace")]
+    backtrace: Option<Backtrace>,
+    #[cfg(feature = "spantrace")]
+    span_trace: Option<SpanTrace>,
+}
+
+impl Report<()> {
+    /// Creates a new `Report` from the provided message.
+    #[track_caller]
+    pub fn new<M>(message: M) -> Self
+    where
+        M: fmt::Display + fmt::Debug + Send + Sync + 'static,
+    {
+        // SAFETY: `FrameRepr` is wrapped in `ManuallyDrop`
+        Self::from_frame(
+            Frame::from_message(message, Location::caller(), None),
+            #[cfg(feature = "backtrace")]
+            Some(Backtrace::capture()),
+            #[cfg(feature = "spantrace")]
+            Some(SpanTrace::capture()),
+        )
+    }
+}
+
+impl<Context> Report<Context> {
+    fn from_frame(
+        frame: Frame,
+        #[cfg(feature = "backtrace")] backtrace: Option<Backtrace>,
+        #[cfg(feature = "spantrace")] span_trace: Option<SpanTrace>,
+    ) -> Self {
+        Self {
+            inner: Box::new(ReportImpl {
+                frame,
+                #[cfg(feature = "backtrace")]
+                backtrace,
+                #[cfg(feature = "spantrace")]
+                span_trace,
+            }),
+            _context: PhantomData,
+        }
+    }
+
+    /// Creates a new `Report<S>` from a provided scope.
+    #[track_caller]
+    pub fn from_context(context: Context) -> Self
+    where
+        Context: Provider + fmt::Display + fmt::Debug + Send + Sync + 'static,
+    {
```

Comment:
As with previous comments, reason for `Report<S>`?,  and can we not just use 
```suggestion
    /// Creates a new `Report<C>` from a provided scope.
    #[track_caller]
    pub fn from_context(context: C) -> Self
    where
        C: Context,
    {
```

Reply:
Same as above (the `Context` thing at least)